build-support/common.sh

```diff
@@ -52,7 +52,15 @@ function determine_python() {
     echo "${PY}"
     return 0
   fi
-  for version in '3.7' '3.8' '3.9'; do
+
+  local candidate_versions
+  if is_macos_arm; then
+    candidate_versions=('3.9')
```

Comment:
My python eyes are screaming "this evaluates to a string, not a tuple!" but in bash we good.

Reply:
Shellcheck was the only way I could figure out how to do this PR - I haven't written bash in a while.